src/transaction/confirm.rs

```diff
@@ -198,19 +192,14 @@ impl<T: Transport> ConfirmationContext<'_, T> {
 
     /// Waits for a certain number of blocks to be mined using a block filter.
     async fn wait_for_blocks_with_filter(&self, block_count: usize) -> Result<(), ExecutionError> {
-        let block_filter = self
-            .web3
-            .eth_filter()
-            .create_blocks_filter()
-            .compat()
-            .await?;
-        let _ = block_filter
-            .stream(self.params.poll_interval)
-            .skip((block_count - 1) as _)
-            .into_future()
-            .compat()
-            .await
-            .map_err(|(err, _)| err)?;
+        let block_filter = self.web3.eth_filter().create_blocks_filter().await?;
+        let mut stream = block_filter.stream(self.params.poll_interval);
+        for _ in 0..block_count {
+            stream
+                .next()
+                .await
+                .ok_or(ExecutionError::StreamEndedUnexpectedly)??;
+        }
```

Comment:
By turning `skip` into a loop we make sure to notice observe all errors.

Reply:
There is also `TryStreamExt::try_skip_while` (but no `try_skip` :confused:) but I think the `for .. in` loop is elegant here.